lib/winappdriver.js

```diff
@@ -130,8 +132,40 @@ class WinAppDriver extends events.EventEmitter {
   }
 
   async startSession (caps) {
+
+    let createSessionTimeout = this.createSessionTimeout;
+    if (caps.createSessionTimeout) {
+      createSessionTimeout = Number.isInteger(caps.createSessionTimeout) ? caps.createSessionTimeout : parseInt(caps.createSessionTimeout, 10);
```

Comment:
this is not needed. if the capability is defined as isNumber, that it will be automatically cast to the necessary type

Reply:
why this comment got resolved without an explanation? the original problem is still there

Reply:
Are you sure this is the case? https://github.com/appium/appium-base-driver/blob/69dde698d6ef33f50b8ec7f88ee0fb0ddc86e1bf/lib/basedriver/commands/session.js#L132 should automatically fix capability types. Do you observe the warning message like https://github.com/appium/appium-base-driver/blob/69dde698d6ef33f50b8ec7f88ee0fb0ddc86e1bf/lib/basedriver/commands/session.js#L142 while running your test and the cap is passed as string?

Reply:
@mykola-mokhnach you was right, the automatic cast has to be, I have debugged the base-driver code and found that desiredCapConstraints is not really updated by subclass and all windows driver specific constraints are ignored. So I added this.desiredCapConstraints = desiredCapConstraints; to the driver constructor. That fixes the issue with the cast.